documentation/projects/proposals/dark_mode/20240325-implementation_plan_dark_mode.md

```diff
@@ -0,0 +1,263 @@
+         not result in any visual changes. **This is likely to be the largest PR
+         to review as it is a global find/replace across the entire frontend.**
+
+         1. Replace the "hardcoded" color values in the Tailwind configuration
+            file with css variables defined in the "base" layer of the
+            `tailwind.css` file. This will not result in any visual changes.
+
+      2. Add the dark mode colors as CSS variable definitions, nested under the
+         `.dark-mode` CSS class, in the "base" layer of the `tailwind.css` file.
+         This will not result in any visual changes, _except when
+         `FORCE_DARK_MODE` is enabled._
+
+      3. Visual Regression tests. Update
+         `frontend/test/playwright/utils/breakpoints.ts` so that each breakpoint
+         produces and expects a dark mode screenshot to pass as well as the
+         existing light mode screenshot. **This will also be a significant diff,
+         as at the time of writing it will create 293 new screenshots to review.
+         This is also the point of the process where @fcoveram and
+         @wordpress/openverse-frontend should review the full dark mode
+         appearance for correctness and sufficient color contrast (see the
+         ["Accessibility"](#accessibility) section for more details.**
+
+2. **Work Stream B**: Toggling dark mode
+
+   1. Create a `DARK_MODE_UI_TOGGLE` feature flag which is off by default and
+      switchable in staging.
+   2. Implement logic for calculating a current "color mode" with the following
+      state:
+
+      ```ts
+      interface ColorMode {
+        preference: "dark" | "light" | "system" // Defaults to "light"
+        systemValue: "light" | "dark" // Readonly representation of the system value
+      }
+      ```
+
+      The color mode should be stored in a cookie so that a previously-selected
+      user choice can be used when rendering via SSR and prevent a visual flash
+      of light mode styles for users who have selected dark mode. The system
+      value is read by matching the `'(prefers-color-scheme: dark)'` media
+      query.
+
+   3. Behind the feature flag, add the new user interface element which toggles
+      dark mode (exact design TBD, but it will be comprised of existing UI
+      components). Default to "light" mode but support choosing between "dark",
+      "light", and "system".
+      1. Add a `TOGGLE_COLOR_SCHEME` analytics event with a playload including
+         the color mode preference chosen by the user.
+
+### Launch plan
+
+> See the ["Rollback"](#rollback) section for details on how to revert this
+> deployment.
+
+1. Set the `DARK_MODE_UI_TOGGLE` feature flag to "on" in all environments.
+2. Deploy the production frontend.
+3. Test and verify the deploy was successful and that dark mode:
+   1. Looks correct
+   2. The toggle works correctly (chosen settings persist, the control works
+      with keyboard, etc.)
```

Comment:
Shouldn't we do this verification step in staging, not production?

Reply:
After the problems we had during launch of additional search views, I think we should verify in staging, and then set the production flag to `enabled` (without ever setting it to `swtichable`)

Reply:
Sounds good!